doc/ch-grow_desc.rst

```diff
@@ -209,16 +180,22 @@ This one is (ironically) :code:`apt-get` failing to drop privileges:
   E: seteuid 100 failed - seteuid (22: Invalid argument)
   E: setgroups 0 failed - setgroups (1: Operation not permitted)
 
-The solution :code:`ch-grow` uses is to intercept these system calls and fake
-a successful result. We accomplish this by altering the Dockerfile to call
-:code:`fakeroot(1)` (of which there are several implementations) for
-:code:`RUN` instructions that seem to need it. There are two basic steps:
+By default, nothing is done to avoid these problems, though :code:`ch-grow`
+does try to detect if the workarounds could help. :code:`--force` activates
+the workarounds: :code:`ch-grow` injects extra commands to intercept these
+system calls and fake a successful result, using :code:`fakeroot(1)`. There
+are three basic steps:
+
+  1. After :code:`FROM`, analyze the image to see what distribution it
+     contains, which determines the specific workarounds.
 
-  1. After :code:`FROM`, install a :code:`fakeroot(1)` implementation. This
-     sometimes also needs extra steps like turning off the :code:`apt` sandbox
-     (for Debian Buster) or enabling EPEL (for CentOS/RHEL).
+  2. Before the user command in the first :code:`RUN` instruction where the
+     injection seems needed, install :code:`fakeroot(1)` in the image, if one
+     is not already installed, as well as any other necessary initialization
+     commands. For example, we turn off the :code:`apt` sandbox (for Debian
+     Buster) and enable EPEL (for CentOS/RHEL).
```

Comment:
Enabling EPEL can have undesirable results, so be careful. If you _only_ need EPEL to install `fakeroot` then I'd suggest:

1. install EPEL repo RPM
1. disable EPEL repository using `yum-config-manager --disable epel`
1. EXPLICITY enable EPEL like `yum --enablerepo=epel`

It's common that there are upstream versions of RPMs in EPEL that will break other non-EPEL distro software. Two weeks ago I had a broken CentOS OpenStack installation. The problem was I had EPEL enabled and it installed some python RPMs that caused the CentOS OpenStack software to fail.

Reply:
You might want to check to see if EPEL is already installed and enabled. That would imply the user knows the EPEL risks and the above is less of a concern.

`yum repolist enabled` will give you the output you want to parse